File: code/sdk/android/cloudsheet-Android-sdk/src/main/kotlin/com/opencloudsheet/auth/IAuthenticator.kt
package com.opencloudsheet.auth

import android.app.Activity
import com.opencloudsheet.model.userdetails.IUserDetails

interface IAuthenticator {
    suspend fun login(activity: Activity): Boolean
    suspend fun getAuthToken(): String?
    suspend fun logout(activity: Activity): Boolean
    suspend fun getUserDetails(): IUserDetails?
}

File: code/sdk/ios/OpenCloudSheet/Sources/OpenCloudSheet/Auth/IAuthenticator.swift
//
//  IAuthenticator.swift
//  Cloudsheet-ios-sdk
//
//  Protocol for cloud storage authentication
//

import Foundation

protocol IAuthenticator {
    func login(from viewController: PlatformViewController) async throws
    func getAuthToken() async throws -> String?
    func getUserDetails() async throws -> IUserDetails?
    func logout() async throws
}
